runtime/vm/resolvefield.cpp

```diff
@@ -1084,62 +1084,76 @@ fieldOffsetsFindNext(J9ROMFieldOffsetWalkState *state, J9ROMFieldShape *field)
 									state->objectsSeen++;
+								}
+							} else {
+								U_32 size = fieldClass->totalInstanceSize;
+								state->result.flattenedClass = fieldClass;
+								if (J9_ARE_ALL_BITS_SET(fieldClass->classFlags, J9ClassLargestAlignmentConstraintDouble)) {
+									if (J9CLASS_HAS_4BYTE_PREPADDING(fieldClass)) {
+										size -= sizeof(U_32);
+									}
+									state->result.offset = state->firstFlatDoubleOffset + state->currentFlatDoubleOffset;
+									Assert_VM_true((state->result.offset + referenceSize) % sizeof(U_64) == 0);
+									state->currentFlatDoubleOffset += ROUND_UP_TO_POWEROF2(size, sizeof(U_64));
+								} else if (J9_ARE_ALL_BITS_SET(fieldClass->classFlags, J9ClassLargestAlignmentConstraintReference)) {
+									size = ROUND_UP_TO_POWEROF2(size, referenceSize);
+									if (J9_ARE_ALL_BITS_SET(state->walkFlags, J9VM_FIELD_OFFSET_WALK_BACKFILL_FLAT_OBJECT_FIELD)
+										&& (state->flatBackFillSize == size)
+									) {
+										Assert_VM_true(state->backfillOffsetToUse >= 0);
+										state->result.offset = state->backfillOffsetToUse;
+										state->walkFlags &= ~(UDATA)J9VM_FIELD_OFFSET_WALK_BACKFILL_FLAT_OBJECT_FIELD;
+									} else {
+										state->result.offset = state->firstFlatObjectOffset + state->currentFlatObjectOffset;
+										state->currentFlatObjectOffset += size;
+									}
 								} else {
-									U_32 size = fieldClass->totalInstanceSize;
-									state->result.flattenedClass = fieldClass;
-									if (J9_ARE_ALL_BITS_SET(fieldClass->classFlags, J9ClassLargestAlignmentConstraintDouble)) {
-										if (J9CLASS_HAS_4BYTE_PREPADDING(fieldClass)) {
-											size -= sizeof(U_32);
-										}
-										state->result.offset = state->firstFlatDoubleOffset + state->currentFlatDoubleOffset;
-										Assert_VM_true((state->result.offset + referenceSize) % sizeof(U_64) == 0);
-										state->currentFlatDoubleOffset += ROUND_UP_TO_POWEROF2(size, sizeof(U_64));
-									} else if (J9_ARE_ALL_BITS_SET(fieldClass->classFlags, J9ClassLargestAlignmentConstraintReference)) {
-										size = ROUND_UP_TO_POWEROF2(size, referenceSize);
-										if (J9_ARE_ALL_BITS_SET(state->walkFlags, J9VM_FIELD_OFFSET_WALK_BACKFILL_FLAT_OBJECT_FIELD)
-											&& (state->flatBackFillSize == size)
-										) {
-											Assert_VM_true(state->backfillOffsetToUse >= 0);
-											state->result.offset = state->backfillOffsetToUse;
-											state->walkFlags &= ~(UDATA)J9VM_FIELD_OFFSET_WALK_BACKFILL_FLAT_OBJECT_FIELD;
-										} else {
-											state->result.offset = state->firstFlatObjectOffset + state->currentFlatObjectOffset;
-											state->currentFlatObjectOffset += size;
-										}
+									if (J9_ARE_ALL_BITS_SET(state->walkFlags, J9VM_FIELD_OFFSET_WALK_BACKFILL_FLAT_SINGLE_FIELD)
+										&& (state->flatBackFillSize == size)
+									) {
+										Assert_VM_true(state->backfillOffsetToUse >= 0);
+										state->result.offset = state->backfillOffsetToUse;
```

Comment:
Given the assert above, I think it would make sense to set `state->backfillOffsetToUse` to 0 here (and in the other similar places).

Reply:
> I think it would make sense to set state->backfillOffsetToUse to 0 here

You mean after line 1129 ?

Reply:
Yes, zero it so the assertion will fire if someone sets the bit without setting the backfill.

Reply:
Is there more than one backfill opportunity? If not, I suppose this doesnt matter, though it is cleaner.

Reply:
> Is there more than one backfill opportunity?

No. I guess it is not necessary to set state->backfillOffsetToUse here.